python/ray/_private/node.py

```diff
@@ -329,9 +329,16 @@ def __init__(
                 self.gcs_address,
                 self._raylet_ip_address,
             )
+            self._node_id = node_info["node_id"]
             if self._ray_params.node_manager_port == 0:
                 self._ray_params.node_manager_port = node_info["node_manager_port"]
 
+        # If it is a head node, try validating if
+        # external storage is configurable.
+        if head:
+            self.validate_external_storage()
+        if not connect_only:
+            self.destroy_external_storage()
```

Comment:
Just realized it won't work any more. The original intention is that we clean up left over spills from the previous Ray node on the same physical machine. But now we have node_id as part of the directory name, we won't be able to know what the previous node_id is and the code is trying to clean up current spill directory which is empty since the node is just started.

I think we can remove these two lines and rely on Ray to clean up upon shutdown for now. cc @rkooo567

Reply:
Yeah this wouldn't have effect for worker nodes.

For head node, it keeps the original behavior of destroying the external storage after validation. I'm going to remove these two lines and add `storage.destroy_external_storage()` to the end of `validate_external_storage()`, which should be cleaner.